src/shared/utils/metrics.ts

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable camelcase */
+import { metrics as sentryMetrics } from '@sentry/react'
+
+type LeafKeyUnion<T, K extends string = ''> = {
+  [P in keyof T]: T[P] extends object
+    ? LeafKeyUnion<T[P], `${K}${K extends '' ? '' : '.'}${string & P}`>
+    : `${K}${K extends '' ? '' : '.'}${string & P}`
+}[keyof T]
+
+type DistributionKeys = {
+  // TODO: Remove this key when adding in non-test keys
+  testKey: string
+}
+
+type GaugeKeys = {
+  // TODO: Remove this key when adding in non-test keys
+  testKey: string
+}
+
+type IncrementKeys = {
+  coverage_tab: {
+    visited_page: string
+  }
+  commit_detail_page: {
+    coverage_page: {
+      visited_page: string
+    }
+    coverage_dropdown: {
+      opened: string
+    }
+    bundle_page: {
+      visited_page: string
+    }
+    bundle_dropdown: {
+      opened: string
+    }
+  }
+  pull_request_page: {
+    coverage_page: {
+      visited_page: string
+    }
+    coverage_dropdown: {
+      opened: string
+    }
+    bundle_page: {
+      visited_page: string
+    }
+    bundle_dropdown: {
+      opened: string
+    }
+  }
+  bundles_tab: {
+    bundle_details: {
+      visited_page: string
+    }
+    onboarding: {
+      visited_page: string
+      copied: {
+        token: string
+        config: string
+        commit: string
+        install_command: string
+        build_command: string
+      }
+    }
+  }
+}
+
+type SetKeys = {
+  // TODO: Remove this key when adding in non-test keys
+  testKey: string
+}
+
+type DistributionValue = Parameters<typeof sentryMetrics.distribution>['1']
```

Comment:
curious as to why we are using strings as indices for collections?

Reply:
The type doesn't really matter in this instance, we're only looking at the key.

Reply:
Oh sorry, if you're talking about for grabbing the value from the `Parameters` TS utility, it's just the way you do it in TS it seems